src/test/java/roomescape/reservation/service/ReservationServiceTest.java

```diff
@@ -0,0 +1,86 @@
+import org.springframework.test.annotation.DirtiesContext;
+import roomescape.reservation.dto.request.ReservationRequest;
+import roomescape.reservation.dto.request.ReservationTimeRequest;
+import roomescape.reservation.dto.response.ReservationResponse;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+class ReservationServiceTest {
+
+    @Autowired
+    private ReservationService reservationService;
+
+    @Autowired
+    private ReservationTimeService reservationTimeService;
+
+    @DisplayName("예약시간이 없는 경우 예외가 발생한다.")
+    @Test
+    void reservationTimeIsNotExist() {
+        ReservationRequest reservationRequest = new ReservationRequest("브라운", LocalDate.of(2023, 8, 5), 1L);
+
+        assertThatThrownBy(() -> reservationService.createReservation(reservationRequest))
+                .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @DisplayName("예약 생성 테스트")
+    @Test
+    void createReservation() {
+        ReservationTimeRequest reservationTimeRequest = new ReservationTimeRequest(LocalTime.of(10, 1));
+        reservationTimeService.createReservationTime(reservationTimeRequest);
+
+        ReservationRequest reservationRequest = new ReservationRequest("브라운", LocalDate.of(2023, 8, 5), 1L);
+        ReservationResponse reservationResponse = reservationService.createReservation(reservationRequest);
+
+        assertAll(
+                () -> assertThat(reservationResponse.name()).isEqualTo("브라운"),
+                () -> assertThat(reservationResponse.date()).isEqualTo(LocalDate.of(2023, 8, 5))
+        );
+    }
+
+    @DisplayName("모든 예약 조회 테스트")
+    @Test
+    void findAllReservations() {
+        ReservationTimeRequest reservationTimeRequest = new ReservationTimeRequest(LocalTime.of(10, 1));
+        reservationTimeService.createReservationTime(reservationTimeRequest);
+
+        ReservationRequest reservationRequest = new ReservationRequest("브라운", LocalDate.of(2023, 8, 5), 1L);
+        reservationService.createReservation(reservationRequest);
+
+        List<ReservationResponse> reservations = reservationService.findAllReservations();
+
+        assertAll(
+                () -> assertThat(reservations).hasSize(1),
+                () -> assertThat(reservations.get(0).name()).isEqualTo("브라운"),
+                () -> assertThat(reservations.get(0).date()).isEqualTo(LocalDate.of(2023, 8, 5))
+        );
+    }
+
+    @DisplayName("예약 삭제 테스트")
+    @Test
+    void deleteReservation() {
+        ReservationTimeRequest reservationTimeRequest = new ReservationTimeRequest(LocalTime.of(10, 1));
+        reservationTimeService.createReservationTime(reservationTimeRequest);
+
+        ReservationRequest reservationRequest = new ReservationRequest("브라운", LocalDate.of(2023, 8, 5), 1L);
+        reservationService.createReservation(reservationRequest);
+
+        reservationService.deleteReservation(1L);
```

Comment:
```suggestion
        Reservation reservation = reservationService.createReservation(reservationRequest);

        reservationService.deleteReservation(reservation.id());
```

저장하고 받은 id를 사용해도 좋겠네요.